### PR TITLE
[IE TESTS] Add Convert, ConvertLike single layer tests. Refactoring ` Range`

### PR DESCRIPTION
**Pull request by iefode**
[IE TESTS] Add Convert, ConvertLike single layer tests. Refactoring ` Range`

### CODE REVIEW

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+#include "single_layer_tests/convert.hpp"
+#include "common_test_utils/test_constants.hpp"
+
+using namespace LayerTestsDefinitions;
+
+namespace {
+
+std::vector<std::vector<std::vector<size_t>>> inShapes = {
+        {{1}},
```
Comment (mikhail-treskin) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert.cpp:15]: As for me looks like extra cases. For convert layer shapes are not really important.

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+std::vector<InferenceEngine::Precision> netPrecisions = {InferenceEngine::Precision::FP32,
+                                                         InferenceEngine::Precision::FP16};
+
+INSTANTIATE_TEST_CASE_P(NoReshape, ConvertLayerTest,
+                        ::testing::Combine(
+                                ::testing::ValuesIn(inShapes),
+                                ::testing::ValuesIn(netPrecisions),
+                                ::testing::ValuesIn(netPrecisions),
```
Comment (mikhail-treskin) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert.cpp:28]: I guess that pretty common scenario its conversion from floating to integer precisions or vice versa, so maybe to add cases with U8 I8 target precisions?

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp [hunk 3]
```diff
@@ -31,12 +31,18 @@ std::vector<std::string> disabledTestPatterns() {
...
         R"(.*RangeLayerTest.*)",
+        R"(.*(RangeAddSubgraphTest).*Start=1.2.*Stop=5.2.*Step=0.1.*netPRC=FP16.*)",
+        // TODO: Issue: 34083
 #if (defined(_WIN32) || defined(_WIN64))
         R"(.*(CoreThreadingTestsWithIterations).*(smoke_LoadNetworkAccuracy).*(targetDevice=HETERO).*)",
 #endif
         // TODO: Issue: 34348
         R"(.*IEClassGetAvailableDevices.*)",
+        // TODO: Issue: 25533
+        R"(.*ConvertLikeLayerTest.*)",
```
Comment (mikhail-treskin) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp:44]: Doesn't important for the PR but just a general comment - from regex perspective braces here (and almost everywhere in existing skips) are redundant

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert_like.cpp [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+#include <vector>
+
+#include "single_layer_tests/convert_like.hpp"
+#include "common_test_utils/test_constants.hpp"
+
+using namespace LayerTestsDefinitions;
+
+namespace {
+
+std::vector<std::vector<std::vector<size_t>>> inShapes = {
```
Comment (mikhail-treskin) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert_like.cpp:14]: Sam comments as for Convert case, but here you produce twice more permutations of shapes for both parameters which in fact will not affect operation behaviour.
Precisions combinations much more important for the operation

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/range_add.cpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+#include <vector>
+
+#include "subgraph_tests/range_add.hpp"
+#include "common_test_utils/test_constants.hpp"
+
+using namespace LayerTestsDefinitions;
+
+namespace {
+
+const std::vector<float> start = { 1.0f, 1.2f };
```
Comment (mikhail-treskin) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/range_add.cpp:14]: maybe to extend with negative numbers in range?

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert.cpp (hunk 1 above)
Comment (iefode) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert.cpp:15]: Agree, TBD

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert.cpp (hunk 2 above)
Comment (iefode) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert.cpp:28]: Agree, TBD

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert_like.cpp (hunk 4 above)
Comment (iefode) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert_like.cpp:14]: Agree, TBD

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/range_add.cpp (hunk 5 above)
Comment (iefode) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/range_add.cpp]: Done

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert_like.cpp (hunk 4 above)
Comment (iefode) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert_like.cpp]: Done

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert.cpp (hunk 2 above)
Comment (iefode) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert.cpp:22]: Done

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert.cpp (hunk 1 above)
Comment (iefode) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/convert.cpp]: Done

**Pull request merged**